arbnode/espresso.go

```diff
@@ -0,0 +1,28 @@
+package arbnode
+
+import (
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/execution"
+	"github.com/offchainlabs/nitro/util/containers"
+)
+
+// TransactionStreamerExecutionSequencer defines the interface under which
```

Comment:
I think this is an unnecessary optimization, I feel like its okay to add another method to the original interface. In future, we will have to manage both the interfaces. I dont see the benefit in that

Reply:
I think so.  We could add a type assertion check of one implementation to the other.  Something like this might work:

```go
var _ TransactionStreamerExecutionSequencer = execution.ExecutionSequencer(nil)
```

Reply:
I will say that explicitly modifying the `TransactionStreamer`'s `exec` field's type is less-than-ideal.  There may even be a middle ground here to introduce another wrapper that adheres to the full `execution.ExecutionClient` with some unimplemented errors for every method call.  Then add a unifying struct that defers to the reduced interface with overrides, but allows the passthrough call to the larger unimplemented case where it's not implemented explicitly.